fdbclient/CMakeLists.txt

```diff
@@ -72,22 +72,22 @@ add_flow_target(STATIC_LIBRARY NAME fdbclient SRCS ${FDBCLIENT_SRCS} ADDL_SRCS $
 target_include_directories(fdbclient PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include" "${CMAKE_CURRENT_BINARY_DIR}/include")
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/versions.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/include/fdbclient/versions.h)
 add_dependencies(fdbclient fdboptions)
-target_link_libraries(fdbclient PUBLIC fdbrpc msgpack PRIVATE rapidxml)
+target_link_libraries(fdbclient PUBLIC fdbrpc msgpack PRIVATE rapidxml rapidjson)
 
 # Create a separate fdbclient library with sampling enabled. This lets
 # fdbserver retain sampling functionality in client code while disabling
 # sampling for pure clients.
 add_flow_target(STATIC_LIBRARY NAME fdbclient_sampling SRCS ${FDBCLIENT_SRCS} ADDL_SRCS ${options_srcs})
 target_include_directories(fdbclient_sampling PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include" "${CMAKE_CURRENT_BINARY_DIR}/include")
 add_dependencies(fdbclient_sampling fdboptions)
-target_link_libraries(fdbclient_sampling PUBLIC fdbrpc_sampling msgpack PRIVATE rapidxml)
+target_link_libraries(fdbclient_sampling PUBLIC fdbrpc_sampling msgpack PRIVATE rapidxml rapidjson)
 target_compile_definitions(fdbclient_sampling PRIVATE -DENABLE_SAMPLING)
 if(WIN32)
   add_dependencies(fdbclient_sampling_actors fdbclient_actors)
 endif()
 
 add_flow_target(LINK_TEST NAME fdbclientlinktest SRCS LinkTest.cpp)
-target_link_libraries(fdbclientlinktest PRIVATE fdbclient rapidxml) # re-link rapidxml due to private link interface
+target_link_libraries(fdbclientlinktest PRIVATE fdbclient rapidxml rapidjson) # re-link rapidxml due to private link interface
```

Comment:
is this change needed because this logic previously lived in the server?
Is there a particular reason that RestKMSConnectorUtils/RESTSimKmsVault should live in the client and not the server, since it's a server-side concept? Besides avoiding this linking change that could potentially increase the size of the fdb client libraries, it seems like a cleaner spot for these function, as the only consumers of this are server-side (RESTKmsConnector/etc...).

Reply:
> is this change needed because this logic previously lived in the server?

Good point, I think I added them to fdbclient, as it allows the module to accessible to both fdbclient and fdbserver if needed, however, I don't think there is any reason for the module to be needed outside fdbserver, hence, can be moved. 
Regarding fdbclient module bloating, it is a valid concern and given current usecase, the module can be contained in fdbserver code and serve intended purpose. 
Thanks